packages/sync/src/offline/OfflineQueue.ts

```diff
@@ -23,79 +27,91 @@ export interface OfflineQueueOptions {
   private readonly storage?: PersistentStore<PersistedData>;
   private readonly storageKey?: string;
   private readonly listener?: OfflineQueueListener;
   private readonly state?: ObjectState;
 
-  constructor(options: OfflineQueueOptions) {
-    super(options);
-
-    const { storage, storageKey, listener, conflictStateProvider } = options;
+  constructor(options: OfflineLinkOptions) {
+    this.storage = options.storage;
+    this.storageKey = options.storageKey;
+    this.listener = options.listener;
+    this.state = options.conflictStateProvider;
+  }
 
-    this.storage = storage;
-    this.storageKey = storageKey;
-    this.listener = listener;
-    this.state = conflictStateProvider;
+  public enqueue(operation: Operation, forward: NextLink) {
+    const operationEntry = new OperationQueueEntry(operation, forward);
+    this.enqueueEntry(operationEntry);
+    return new Observable((observer) => {
+      operationEntry.observer = observer;
+      return () => ({ isOffline: true });
+    });
   }
 
-  /**
-   * Returns list of operations that can be forwarded - i.e. they have not
-   * been forwarded yet and do not have client ID.
-   */
-  public toBeForwarded() {
-    return this.queue.filter(op => !op.subscription && !op.hasClientId());
+  public async forwardOperations() {
+    for (const op of this.queue) {
+      // FIXME block operations till result is back (completed)
+      await new Promise((resolve, reject) => {
+        op.forward(op.operation).subscribe({
+          next: (result: FetchResult) => {
+            this.queue = this.queue.filter(e => e !== op);
+            if (result.errors) {
+              if (this.listener && this.listener.onOperationFailure) {
+                this.listener.onOperationFailure(op.operation, result.errors);
+              }
+              // Notify for success otherwise
+            } else if (result.data) {
+              if (this.listener && this.listener.onOperationSuccess) {
+                this.listener.onOperationSuccess(op.operation, result.data);
+              }
+              this.updateIds(op, result);
+              this.updateObjectState(op, result);
+            }
+            this.persist();
+
+            if (this.queue.length === 0 && this.listener && this.listener.queueCleared) {
+              this.listener.queueCleared();
+            }
+            if (op.observer) {
+              op.observer.next(result);
+            }
+          },
+          error: (error: any) => {
+            // TODO remove result from operation and handle that directly.
+            if (error) {
+              if (this.listener && this.listener.onOperationFailure) {
+                this.listener.onOperationFailure(op.operation, undefined, op.networkError);
+              }
+              return;
+            }
+            if (op.observer) {
+              op.observer.error(error);
+            }
+            // TODO - do we want to handle try catch here?
```

Comment:
Did I wrote that. If yes then let's remove it as I do not know what `handle try catch` can mean in this context. There is no error thrown.